tests/compiler_test.go

```diff
@@ -0,0 +1,17 @@
+// +build js
+
+package tests
+
+import (
+	"testing"
+)
+
+func TestVariadicNil(t *testing.T) {
+	printVari := func(strs ...string) []string {
+		return strs
+	}
+
+	if v := printVari(); v != nil {
+		t.Errorf("expected printVari() to be %v; got: %v", nil, v)
+	}
+}
```

Comment:
Isn't `tests/misc_test.go` a proper place?

Reply:
The `tests` package needs a bit of a cleanup at some point. The segmentation is, I agree, not great. I even be tempted to have subpackages to better segment things. 

But that belongs in a later PR. Not critical either way in my opinion whether this goes in `misc_test.go` or a new file.

Reply:
I was wondering why `compiler_test.go` didn't exist until this PR.